### PR TITLE
Travis-CI improvements

### PR DESCRIPTION
**Pull request by rcchen**
Adding `sudo: false` switches us over to Travis' container-based infrastructure, which should give us modest performance boosts in testing.

### CODE REVIEW

**Comment by aicioara** [comment]
Can you provide a URL with the documentation? It's hard to review a config file otherwise.

**Comment by aicioara** [comment]
there is still an `npm test` which takes about 2 mins out of our 3 min build time. Can we maybe get rid of that as well? It would be so cool

**Comment by rcchen** [comment]
http://docs.travis-ci.com/user/migrating-from-legacy/?utm_source=legacy-notice&utm_medium=banner&utm_campaign=legacy-upgrade

**Comment by rcchen** [comment]
If you run `npm test` on your local machine, you'll notice that it encompasses our test suite. I don't think it's possible to throw that out, unless there's a redundancy somewhere that I'm not noticing.

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,14 @@
 language: node_js
 node_js:
-- '0.10'
+- '0.12'
+before_install:
+- phantomjs -v
 notifications:
   email: false
+sudo: false
```
Comment (aicioara) [inline .travis.yml]: This is awesome news. It says that we are guaranteed <1 min queue time? Is that correct? If it is, that's amazing.

**Comment by rcchen** [comment]
The benefit that we're mostly going to be realizing from this is that boot times for machines is reduced (time to start of test)

File: .travis.yml (hunk 1 above)
Comment (rcchen) [inline .travis.yml]: We are guaranteed < 10 sec build starts, and more consistent builds throughout the day :smiley:

**Comment by aicioara** [comment]
Does boot time mean queue + boot time?

**Comment by aicioara** [comment]
Let's merge this and watch it more in the wild. The idea sounds awesome to me.

**Comment by rcchen** [comment]
Queue time is a function of the number of users at any given time, and how fast their builds are going/how fast things are set up/torn down, so queue is going to be variable but overall this should help out.

**Pull request merged**